src/main/java/com/microsoft/sqlserver/jdbc/AuthenticationJNI.java

```diff
@@ -51,22 +58,68 @@ static boolean isDllLoaded() {
         try {
             System.loadLibrary(SQLServerDriver.AUTH_DLL_NAME);
             int[] pkg = new int[1];
-            pkg[0] = 0;
+
             if (0 == SNISecInitPackage(pkg, authLogger)) {
                 sspiBlobMaxlen = pkg[0];
             } else {
                 throw new UnsatisfiedLinkError();
             }
             enabled = true;
-        } catch (UnsatisfiedLinkError e) {
-            MessageFormat form = new MessageFormat(SQLServerException.getErrString("R_UnableLoadAuthDll"));
-            temp = new UnsatisfiedLinkError(form.format(new Object[] {SQLServerDriver.AUTH_DLL_NAME}));
-            // This is not re-thrown on purpose - the constructor will terminate the properly with the appropriate error
-            // string
+        } catch (UnsatisfiedLinkError ue) {
+            // If os is windows, attempt to extract and load the DLL packaged with the driver
+            if (SQLServerConnection.isWindows) {
+
+                String tempDirectory = System.getProperty("java.io.tmpdir") + SQLServerDriver.DLL_NAME + "\\";
+                File outputDLL = new File(tempDirectory + UUID.randomUUID() + "-" + new Date().getTime() + ".dll");
+
+                try {
+                    Files.createDirectories(Paths.get(tempDirectory));
+
+                    byte[] buffer = new byte[1024];
+                    try (InputStream is = AuthenticationJNI.class.getResourceAsStream(
+                            "/" + SQLServerDriver.DLL_NAME + "." + Util.getJVMArchOnWindows() + ".dll")) {
+                        try (FileOutputStream fos = new FileOutputStream(outputDLL)) {
+                            int read;
+                            if (is != null) {
+                                while ((read = is.read(buffer)) != -1) {
+                                    fos.write(buffer, 0, read);
+                                }
+                            }
+                        }
+                    }
+
+                    System.load(outputDLL.getAbsolutePath());
+
+                    int[] pkg = new int[1];
+
+                    if (0 == SNISecInitPackage(pkg, authLogger)) {
+                        sspiBlobMaxlen = pkg[0];
+                    } else {
+                        throw new UnsatisfiedLinkError();
+                    }
+
+                    enabled = true;
+
+                } catch (Exception e) {

```

Comment:
instead of catching Exception, can we catch specific exception so we can have more helpful error messages?